test/command-line/execution/executors-test.cpp

```diff
@@ -170,6 +175,94 @@ TEST(ExecutorsTest, TestDistancePipelineExecutor) {
     std::remove(temp_df);
 }
 
+// new test using the no-filter constructor
+TEST(ExecutorsTest, TestDistancePipelineExecutor_NoFilters) {
```

Comment:
I dislike the use of `_` within test case functions, please don't use underscores, just do capitalized naming conventions like the other test cases.